<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Whisker_Widget</class>
 <widget class="QMainWindow" name="Whisker_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>327</width>
    <height>573</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Whisker Tracing</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="trace_button">
          <property name="text">
           <string>Trace</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_mask_mode">
        <item>
         <widget class="QCheckBox" name="use_mask_checkbox">
          <property name="text">
           <string>Use Mask Data</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="mask_key_combo">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Select MaskData key</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_22">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Num Frames to Trace</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="num_frames_to_trace">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_whisker_pad">
          <property name="text">
           <string>Whisker Pad:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="whisker_pad_combo">
          <property name="toolTip">
           <string>Select existing PointData or create new whisker pad</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_whisker_pad_frame">
        <item>
         <widget class="QLabel" name="label_whisker_pad_frame">
          <property name="text">
           <string>Frame:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="whisker_pad_frame_spinbox">
          <property name="toolTip">
           <string>Select which frame from the PointData to use</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>999999</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="whisker_pad_pos_label">
          <property name="text">
           <string>(0,0)</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Length Threshold</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="length_threshold_spinbox">
          <property name="maximum">
           <double>600.000000000000000</double>
          </property>
          <property name="value">
           <double>75.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QComboBox" name="face_orientation">
          <item>
           <property name="text">
            <string>Facing Top</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Facing Bottom</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Facing Left</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Facing Right</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Head Orientation</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QSpinBox" name="whisker_number">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Number of Whiskers</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QSpinBox" name="whisker_clip">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Clip Whisker Length</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_13">
        <item>
         <widget class="QSpinBox" name="manual_whisker_select_spinbox">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>0</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>Manual Whisker Select</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_16">
        <item>
         <widget class="QSpinBox" name="linking_tol_spinbox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>400</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>Linking Tol</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QPushButton" name="dl_trace_button">
          <property name="text">
           <string>DL Trace</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="dl_add_memory_button">
          <property name="text">
           <string>Add Memory Frame</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="auto_dl_checkbox">
          <property name="text">
           <string>Auto</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>327</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuWhisker">
    <property name="title">
     <string>Whisker</string>
    </property>
    <widget class="QMenu" name="menuLoad_Whiskers">
     <property name="title">
      <string>Load Whiskers</string>
     </property>
     <addaction name="actionLoad_Janelia_Whiskers"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="menuLoad_Whiskers"/>
    <addaction name="actionJanelia_Settings"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <widget class="QMenu" name="menuFace_Mask">
     <property name="title">
      <string>Face Mask</string>
     </property>
     <addaction name="actionLoad_Face_Mask"/>
    </widget>
    <addaction name="menuFace_Mask"/>
   </widget>
   <addaction name="menuImage"/>
   <addaction name="menuWhisker"/>
  </widget>
  <action name="actionOpen_Contact_Detection">
   <property name="text">
    <string>Open Contact Detection</string>
   </property>
  </action>
  <action name="actionMask">
   <property name="text">
    <string>Load Mask</string>
   </property>
  </action>
  <action name="actionLoad_Janelia_Whiskers">
   <property name="text">
    <string>Load Janelia Whiskers</string>
   </property>
  </action>
  <action name="actionSave_Snapshot">
   <property name="text">
    <string>Save Snapshot</string>
   </property>
  </action>
  <action name="actionJanelia_Settings">
   <property name="text">
    <string>Janelia Settings</string>
   </property>
  </action>
  <action name="actionLoad_CSV_Whiskers">
   <property name="text">
    <string>Load CSV Whiskers (Single File, Single Frame)</string>
   </property>
  </action>
  <action name="actionLoad_CSV_Whiskers_Multiple">
   <property name="text">
    <string>Load CSV Whiskers (Single File, Single Frame, Multiple)</string>
   </property>
  </action>
  <action name="actionSave_Face_Mask_2">
   <property name="text">
    <string>Save Face Mask</string>
   </property>
  </action>
  <action name="actionLoad_Face_Mask">
   <property name="text">
    <string>Load Face Mask</string>
   </property>
  </action>
  <action name="actionLoad_Mask">
   <property name="text">
    <string>Load HDF5 Mask (Single)</string>
   </property>
  </action>
  <action name="actionLoad_Keypoint_CSV">
   <property name="text">
    <string>Load Follicle (CSV)</string>
   </property>
  </action>
  <action name="actionLoad_HDF5_Whisker_Single">
   <property name="text">
    <string>Load HDF5 Whisker (Single)</string>
   </property>
  </action>
  <action name="actionLoad_HDF5_Whisker_Multiple">
   <property name="text">
    <string>Load HDF5 Whisker (Multiple)</string>
   </property>
  </action>
  <action name="actionLoad_HDF5_Mask_Multiple">
   <property name="text">
    <string>Load HDF5 Mask (Multiple)</string>
   </property>
  </action>
  <action name="actionSave_as_CSV">
   <property name="text">
    <string>Save as CSV</string>
   </property>
  </action>
  <action name="actionLoad_CSV_Whisker_Single_File_Multi_Frame">
   <property name="text">
    <string>Load CSV Whisker (Single File, Multi Frame)</string>
   </property>
  </action>
  <action name="actionSave_as_Binary">
   <property name="text">
    <string>Save as Binary</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
